diff --git a/shell/resources/pps-document-view.blp b/shell/resources/pps-document-view.blp
index b839b7ce0..c008bc2e8 100644
--- a/shell/resources/pps-document-view.blp
+++ b/shell/resources/pps-document-view.blp
@@ -197,6 +197,21 @@ template $PpsDocumentView: Adw.BreakpointBin {
           tooltip-text: _("Toggle Sidebar");
         }
 
+        [start]
+        Button print_shortcut_button {
+          child: Box {
+            spacing: 6;
+            Image {
+              icon-name: "printer-symbolic";
+            }
+            Label {
+              label: _("Print…");
+            }
+          };
+          action-name: "doc.print";
+          tooltip-text: _("Print…");
+        }
+
         [end]
         MenuButton document_menu_button {
           icon-name: "view-more-symbolic";
