diff --git a/shell/pps-document-view.ui b/shell/pps-document-view.ui
index 6c1c89cea..c8e59fead 100644
--- a/shell/pps-document-view.ui
+++ b/shell/pps-document-view.ui
@@ -211,6 +211,27 @@
                     <property name="tooltip-text" translatable="yes">Toggle Sidebar</property>
                   </object>
                 </child>
+                <child type="start">
+                  <object class="GtkButton">
+                    <property name="child">
+                      <object class="GtkBox">
+                        <property name="spacing">6</property>
+                        <child>
+                          <object class="GtkImage">
+                            <property name="icon-name">printer-symbolic</property>
+                          </object>
+                        </child>
+                        <child>
+                          <object class="GtkLabel">
+                            <property name="label" translatable="yes">Print…</property>
+                          </object>
+                        </child>
+                      </object>
+                    </property>
+                    <property name="tooltip_text" translatable="yes">Print…</property>
+                    <property name="action-name">win.print</property>
+                  </object>
+                </child>
                 <child type="end">
                   <object class="GtkMenuButton" id="document_menu_button">
                     <property name="icon-name">view-more-symbolic</property>
